Modix/Modules/ChannelViewerModule.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+
+namespace Modix.Modules
+{
+    [Group("channel"), Name("Channel Viewer"), Summary("Allows you to show and hide specific channels.")]
+    public class ChannelViewerModule : ModuleBase
+    {
+        [Command("hide")]
+        public async Task HideChannelAsync(IGuildChannel channel)
+        {
+            await channel.AddPermissionOverwriteAsync(Context.User,
+                new OverwritePermissions(readMessages: PermValue.Deny, sendMessages: PermValue.Deny));
+            await ReplyAsync("Channel hidden.");
+        }
+
+        [Command("show")]
+        public async Task ShowChannelAsync(IGuildChannel channel)
+        {
+            await channel.AddPermissionOverwriteAsync(Context.User,
+                new OverwritePermissions(readMessages: PermValue.Allow, sendMessages: PermValue.Allow));
+            await ReplyAsync("Channel hidden.");
+        }
```

Comment:
Has this been tested recently? I had to scuttle a chunk of the Moderation feature that deals with permission overwrites for roles within channels (for muting) because Discord.NET has an outstanding bug where it throws `NullReferenceException` any time I try to all a `PermissionOverwrite` related method. It's supposedly been fixed in the latest version, but they're not gonna fix it in 1.0.2 in any form.